### ISSUE TITLE
Should getSubdomains return an empty array if no subdomains found?

### ISSUE THREAD

**Issue by Rohaq** [layershifter/TLDExtract#27]
As part of some validation requirements, I wanted to loop over subdomain labels, and since an array was the expected return of the `getSubdomains()` function, I assumed it would return an empty array if none were found.

Turns out it returns a `null`, meaning it broke any attempts to use it as the expected return type. I ended up working around this using a null coalescence like the following:

```php
$subdomainLabels = $hostnameExtract->getSubdomains() ?? array();
```

So, should this function return an empty array by default? I'd personally prefer if it did, since it keeps the return type consistent - but others might have differing opinions.

**Comment by layershifter** [comment]
I fully agree with you, `getSubdomains()` should always return an array. Good enchantment for next major.

**Comment by Rohaq** [comment]
Good stuff, it might also be worth digging around to see if there are any others that do the same.

**Comment by layershifter** [comment]
@Rohaq thank you, released in `2.0.0`.
